torch/csrc/autograd/profiler_kineto.cpp

```diff
@@ -268,6 +274,10 @@ void pushProfilingCallbacks() {
           }
           ctx_ptr->stack = callstackStr(cs);
         }
+        if (config.with_module_hierarchy &&
+            fn.scope() != at::RecordScope::BACKWARD_FUNCTION) {
+          ctx_ptr->module_hierarchy = jit::currentModuleHierarchy();
```

Comment:
Can the main part of the work actually be moved to finalizeCPUTrace()?

Reply:
Why the check for backward function here and above?

Reply:
> Why the check for backward function here and above?

Good question. For now I just follow other pattern for stacktrace. I am not 100% sure if same information is available for backward functions.

Reply:
> Can the main part of the work actually be moved to finalizeCPUTrace()?

Let me check. It has same constraints as `jit::currentStackTrace`. By the time we go to finalizeCPUTrace scope of JIT is gone and we wont get any information.

For the purpose of reducing overhead, we may have to do it differently, but I havent yet thought about it.